lyncs_io/hdf5.py

```diff
@@ -101,12 +145,41 @@ def split_key(key):
     return "/" + "/".join(tmp[:-1]), tmp[-1]
 
 
-def save(data, filename, key=None, **kwargs):
+def _write(h5f, data, key, **kwargs):
+    "Ensures a group is created and the dataset is written"
+    group, dataset = split_key(key)
+    h5f = h5f.require_group(group)
+    return _write_dataset(h5f, dataset, data, **kwargs)
+
+
+def _write_dispatch(h5f, data, key, **kwargs):
+    if isinstance(data, Mapping):
+        for map_key, val in data.items():
+            _write_dispatch(h5f, val, key + "/" + map_key, **kwargs)
+    else:
+        _write(h5f, data, key, **kwargs)
+
+
+def save(data, filename, key=None, comm=None, **kwargs):
     "Save function for HDF5"
     filename, key = split_filename(filename, key)
     key = key or "/"
+    # Append comm in kwargs
+    kwargs = {"comm": comm, **kwargs}
+
+    if is_dask_array(data):
+        raise NotImplementedError("DaskIO for HDF5 save not implemented yet.")
+
+    if comm is not None:
+        check_comm(comm)
+
+        if comm.size > 1:
+            try:
```

Comment:
Add a "TODO" to restore only the case with "a" when the issue in h5py is fixed

Reply:
I have added the TODO. Shall I open an issue instead so that we keep an eye on it and not forget about it?

Reply:
I should get notified when PR gets merged, so it's ok like this, I will try to remember :)